### PR TITLE
fix(eventindexer): fix chartByTask query bindings and cast to date for start & end date in chart

### PR DESCRIPTION
**Pull request by arif-x**
#16276

### CODE REVIEW

**Comment by bigrayfx** [comment]
yes can't do it

**Comment by cyberhorsey** [comment]
sorry, what do we need this for?
Also this PR doesn't update bindings but it says it does in the title.

**Comment by arif-x** [comment]
> sorry, what do we need this for? is the chart API not working as expected? Also this PR doesn't update bindings but it says it does in the title.
I just knowing when I call chartByTask endpoint, I try to check this query and binding. Because swagger not have start and end params, so this query cant run

**Comment by arif-x** [comment]
> sorry, what do we need this for? is the chart API not working as expected? Also this PR doesn't update bindings but it says it does in the title.

I just knowing when I call chartByTask endpoint, I try to check this query and binding. Because swagger not have start and end params, so this query cant run.
I try to update that query with if start and end param is not set condition, and this query can fecth data. And because date column is using varchar, I try to parse date column into date datatype in select statement.

**Comment by arif-x** [comment]
Sorry my english is bad

File: packages/eventindexer/docs/swagger.json [hunk 1]
```diff
@@ -1,311 +1,359 @@
...
-    },
-    "eventindexer.Stat": {
-      "type": "object",
-      "properties": {
-        "averageProofReward": {
-          "type": "string"
-        },
-        "averageProofTime": {
-          "type": "string"
+    "swagger": "2.0",
```
Comment (xiaodino) [inline packages/eventindexer/docs/swagger.json:2]: Is this change not related?

File: packages/eventindexer/docs/docs.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ const docTemplate = `{
                 }
             }
         },
-        "/chartByTask": {
+        "/chart/chartByTask": {
```
Comment (xiaodino) [inline packages/eventindexer/docs/docs.go:55]: Do we also need to change this https://github.com/taikoxyz/taiko-mono/blob/main/packages/eventindexer/pkg/http/routes.go#L25 ?

**Review by xiaodino — CHANGES_REQUESTED** [review]
What is the motivation for this PR?

File: packages/eventindexer/pkg/repo/chart.go [hunk 3]
```diff
@@ -38,43 +40,80 @@ func (r *ChartRepository) Find(
...
+	}
 
-	tx = r.getDB().Raw(q, task, start, end)
+	var q string
+	var args []interface{}
+	if feeTokenAddress == "" && tier == "" {
+		if start != "" && end != "" {
+			q = `SELECT * FROM time_series_data
+			WHERE task = ? AND STR_TO_DATE(date, '%Y-%m-%d') BETWEEN ? AND ?
+			ORDER BY STR_TO_DATE(date, '%Y-%m-%d')`
```
Comment (xiaodino) [inline packages/eventindexer/pkg/repo/chart.go:66]: Does it mean the query will only query by date?

File: packages/eventindexer/pkg/repo/chart.go [hunk 4]
```diff
@@ -38,43 +40,80 @@ func (r *ChartRepository) Find(
...
+		}
+	}
 
-	tx = r.getDB().Raw(q, task, start, end)
+	var q string
+	var args []interface{}
+	if feeTokenAddress == "" && tier == "" {
+		if start != "" && end != "" {
+			q = `SELECT * FROM time_series_data
+			WHERE task = ? AND STR_TO_DATE(date, '%Y-%m-%d') BETWEEN ? AND ?
```
Comment (xiaodino) [inline packages/eventindexer/pkg/repo/chart.go:65]: Does it mean the query will only filter data by date?

File: packages/eventindexer/pkg/repo/chart.go (hunk 3 above)
Comment (xiaodino) [inline packages/eventindexer/pkg/repo/chart.go:66]: Why use ```ORDER BY STR_TO_DATE(date, '%Y-%m-%d')```, rather than ```ORDER BY date```?

**Comment by cyberhorsey** [comment]
This chart code does not need to be changed, and runs fine in production already. All we need to do is update the documentation, as yes, the route is `/chart/chartByTask`, which is outdated in swagger.

However, this is only used for our status page.

File: packages/eventindexer/pkg/repo/chart.go (hunk 3 above)
Comment (arif-x) [reply packages/eventindexer/pkg/repo/chart.go:66]: because in native mysql need to be casted at date data type

File: packages/eventindexer/pkg/repo/chart.go (hunk 4 above)
Comment (arif-x) [reply packages/eventindexer/pkg/repo/chart.go:65]: I add option if start & end date is filled or not

Comment (arif-x) [reply packages/eventindexer/pkg/repo/chart.go:65]: I add option if start & end date is filled or not

File: packages/eventindexer/docs/docs.go (hunk 2 above)
Comment (arif-x) [reply packages/eventindexer/docs/docs.go:55]: I change this because on https://eventindexer-swagger.taiko.xyz/#/ has not found result
